ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,252 @@
+from collections import defaultdict
+from typing import Callable, cast, Dict, List, Optional, Tuple, Union
+
+import torch
+from torch.nn import functional as F
+
+import ignite.distributed as idist
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+
+class MeanAveragePrecision(Metric):
+    def __init__(
+        self,
+        iou_thresholds: Optional[Union[List[float], torch.Tensor]] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+        r"""Calculate the mean average precision of overall categories.
+
+        Args:
+            iou_thresholds: list of IoU thresholds to be considered for computing Mean Average Precision.
+                Values should be between 0 and 1. List is sorted internally. Default is that of the COCO
+                official evaluation metric.
+            output_transform: a callable that is used to transform the
+                :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+                form expected by the metric. This can be useful if, for example, you have a multi-output model and
+                you want to compute the metric with respect to one of the outputs.
+                By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+            device: specifies which device updates are accumulated on. Setting the
+                metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+                non-blocking. By default, CPU.
+        """
+        try:
+            from torchvision.ops import box_iou
+
+            self.box_iou = box_iou
+        except ImportError:
+            raise ModuleNotFoundError("This module requires torchvision to be installed.")
+
+        if iou_thresholds is None:
+            self.iou_thresholds = [
```

Comment:
Can we replace this with `torch.linspace(0.5, 0.95, 10, device=device, dtype=torch.double)` ?

Reply:
Yes. I did and even I could decrease `tol` from `1e-3` to `1e-4`.